Source/RN4UE4/ReplicaRigidDynamicClient.cpp

```diff
@@ -40,19 +167,70 @@ RigidDynamicConstructionData UReplicaRigidDynamicClient::GetConstructionData()
 
 	FMatrix conversionMatrix = FMatrix();
 	memcpy(conversionMatrix.M, matrixElements, 16 * sizeof(float));
-
+	actorTransform.SetScale3D(FVector(-actorTransform.GetScale3D().X, actorTransform.GetScale3D().Y, actorTransform.GetScale3D().Z));
 	actorTransform *= FTransform(conversionMatrix.Inverse());
 	actorTransform.ScaleTranslation(1 / 50.0f);
 
-	RigidDynamicConstructionData data;
-	data.geom = 0;
-	data.pos.X = -actorTransform.GetLocation().X;
+	GetNearestStaticMesh();
+	ReadPhysicValues(data);
+
+	FTransform centerMassTransform = FTransform(FQuat(data.centerMassRot.X, data.centerMassRot.Y, data.centerMassRot.Z, data.centerMassRot.W), FVector(data.centerMass.X, data.centerMass.Y, data.centerMass.Z), FVector(-1.0f, 1.0f, 1.0f));
+	centerMassTransform *= FTransform(conversionMatrix.Inverse());
+	centerMassTransform.ScaleTranslation(1 / 50.0f);
+
+	data.clientCreated = true;
+	data.pos.X = actorTransform.GetLocation().X;
```

Comment:
Can we simplify to `data.pos = actorTransform.GetLocation()`? (and sample below for `rot` and `scale`)